adapters/repos/db/search.go

```diff
@@ -176,6 +179,71 @@ func hydrateObjectsIntoSearchResults(objs []*storobj.Object, dists []float32) se
 	return res
 }
 
+func (d *DB) ReferenceVectorSearch(ctx context.Context,
+	obj *models.Object, refProps map[string]struct{},
+) ([][]float32, error) {
+	return d.getReferenceVectorsFromParent(ctx, obj, refProps)
+}
+
+func (d *DB) getReferenceVectorsFromParent(ctx context.Context,
+	parent *models.Object, referencePropNames map[string]struct{},
+) (refVecs [][]float32, err error) {
```

Comment:
Actually by defining the result variables in the method, now in line 198 there's this strange assignment (`err = findErr`) just to make use of those variables...

Reply:
Fixed